File: api-surface/src/services/api-security/quantityUsageOrderController.ts
// @ts-ignore
/* eslint-disable */
import {request} from '@umijs/max';

/** 此处后端没有提供注释 POST /order/page */
export async function viewQuantityUsageOrderPage(
  body: API.QuantityUsageOrderPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultQuantityUsageOrderPageVO>(`/security/order/page`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /order/quantity/usage/cancellation */
export async function cancelQuantityUsageOrder(
  body: API.QuantityUsageOrderCancellationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/order/quantity/usage/cancellation`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /order/quantity/usage/confirmation */
export async function confirmQuantityUsageOrder(
  body: API.QuantityUsageOrderConfirmationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/order/quantity/usage/confirmation`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /order/quantity/usage/creation */
export async function placeQuantityUsageOrder(
  body: API.QuantityUsageOrderCreationDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/order/quantity/usage/creation`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: api-security/src/main/java/pers/project/api/security/controller/QuantityUsageOrderController.java
package pers.project.api.security.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pers.project.api.common.model.Result;
import pers.project.api.common.util.ResultUtils;
import pers.project.api.security.model.dto.QuantityUsageOrderCancellationDTO;
import pers.project.api.security.model.dto.QuantityUsageOrderConfirmationDTO;
import pers.project.api.security.model.dto.QuantityUsageOrderCreationDTO;
import pers.project.api.security.model.query.QuantityUsageOrderPageQuery;
import pers.project.api.security.model.vo.QuantityUsageOrderPageVO;
import pers.project.api.security.service.QuantityUsageOrderService;

/**
 * 接口计数用法订单控制器
 *
 * @author Luo Fei
 * @date 2023/07/08
 */
@Validated
@RestController
@RequiredArgsConstructor
@RequestMapping("/order")
public class QuantityUsageOrderController {

    private final QuantityUsageOrderService userOrderService;

    @PostMapping("/page")
    public Result<QuantityUsageOrderPageVO> viewQuantityUsageOrderPage
            (@Valid @RequestBody QuantityUsageOrderPageQuery pageQuery) {
        QuantityUsageOrderPageVO pageVO = userOrderService.getQuantityUsageOrderPageVO(pageQuery);
        return ResultUtils.success(pageVO);
    }

    @PostMapping("/quantity/usage/creation")
    public Result<Void> placeQuantityUsageOrder
            (@Valid @RequestBody QuantityUsageOrderCreationDTO creationDTO) {
        userOrderService.createQuantityUsageOrder(creationDTO);
        return ResultUtils.success();
    }

    @PostMapping("/quantity/usage/cancellation")
    public Result<Void> cancelQuantityUsageOrder
            (@Valid @RequestBody QuantityUsageOrderCancellationDTO cancellationDTO) {
        userOrderService.cancelQuantityUsageOrder(cancellationDTO);
        return ResultUtils.success();
    }

    @PostMapping("/quantity/usage/confirmation")
    public Result<Void> confirmQuantityUsageOrder
            (@Valid @RequestBody QuantityUsageOrderConfirmationDTO orderConfirmationDTO) {
        userOrderService.confirmQuantityUsageOrder(orderConfirmationDTO);
        return ResultUtils.success();
    }

}
